executors/cpp/coll.cpp

```diff
@@ -292,12 +346,28 @@ auto TestCollator(json_object *json_in) -> string {
         this_locale = Locale(locale_string);
       }
       uni_coll = Collator::createInstance(this_locale, status);
+      if (check_icu_error(status, return_json,
+                          "Collator:createInstance")) {
+        json_object_object_add(
+            return_json, "actual_options",
+            json_object_new_string(json_object_get_string(json_in)));
+
+        return json_object_to_json_string(return_json);
+      }
     }
 
-    if (reorder_obj) {
-      if (debug_level > 0) {
-        cout << "# UNI_COLL: reorder codes: " << reorder_string << "(" << reorder_codes_v.size() << ")" << endl;
+    if (alternate_obj) {
+      uni_coll->setAttribute(UCOL_ALTERNATE_HANDLING, alternate_value, status);
+      if (check_icu_error(status, return_json, "alternate")) {
+        json_object_object_add(
+            return_json, "actual_options",
+            json_object_new_string(json_object_get_string(json_in)));
+
+        return json_object_to_json_string(return_json);
       }
+    }
```

Comment:
What's the difference between this and lines 290-299?

Reply:
It appears that the first instance is gated by `if (rules_len > 0)` which uses a different code path to configure the runtime-rules collator. It would probably clean up the code to do more code sharing between these two code paths, but it is a pre-existing issue.

Reply:
Yes, that's right, Shane. This needs to be refactored. I'll add an issue.